### ISSUE TITLE
PERMANOVA on filtered DistanceMatrix returns strange results

### ISSUE THREAD

**Issue by sjanssen2** [biocore/scikit-bio#1877]
I stumble an unexpected behaviour with PERMANOVA computation if I filter the provided DistanceMatrix. It looks like PERMANOVA results are fundamentally different (considering `test statistic` and `p-value`) between an unfiltered distance matrix and a pseudo-filtered one - I use the term pseudo-filtered, because I am using the `.filter()` method, but provide an ID list which should NOT filter out a single element (compare `sample size`).
Is this an error? If so, in the filter method (which claims to[ preserve element ordering](http://scikit-bio.org/docs/latest/generated/skbio.stats.distance.DistanceMatrix.filter.html#skbio.stats.distance.DistanceMatrix.filter)) or in PERMANOVA. The documentation speaks about `pd.DataFrame` for `grouping` and that "[the order of IDs between distance_matrix and the DataFrame need not be the same.](http://scikit-bio.org/docs/latest/generated/skbio.stats.distance.permanova.html#skbio.stats.distance.permanova)".

Any idea what is going wrong?

I just created a fresh conda environment via `mamba create --name skbio -c conda-forge scikit-bio` on my ubuntu 20.04 machine `Linux lummerland 5.4.0-150-generic #167-Ubuntu SMP Mon May 15 17:35:05 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux`. Example data is attached.
My conda env lists the following skbio version:
```scikit-bio                            0.5.9  py311h1f0f07a_0          conda-forge/linux-64```

load modules and read data
```
In [1]: import skbio
In [2]: import pandas as pd
In [3]: dm = skbio.stats.distance.DistanceMatrix.read('example.dm')
In [4]: meta = pd.read_csv("example.meta.csv", sep="\t", index_col=0)
```
expected behaviour:
```
In [5]: skbio.stats.distance.permanova(dm, meta['tumor'])
Out[5]: 
method name               PERMANOVA
test statistic name        pseudo-F
sample size                      42
number of groups                  2
test statistic             4.116198
p-value                       0.008
number of permutations          999
Name: PERMANOVA results, dtype: object
```
**unexpected** behaviour:
```
Out[11]: 
method name               PERMANOVA
test statistic name        pseudo-F
sample size                      42
number of groups                  2
test statistic             1.076027
p-value                       0.346
number of permutations          999
Name: PERMANOVA results, dtype: object
```

[example.dm.txt](https://github.com/biocore/scikit-bio/files/12612628/example.dm.txt)
[example.meta.csv](https://github.com/biocore/scikit-bio/files/12612629/example.meta.csv)

**Comment by wasade** [comment]
Hi @sjanssen2,

That is puzzling. We're taking a look right now.

Best,
Daniel

**Comment by sjanssen2** [comment]
hm, maybe the issue is not with filtering but how the grouping is presented? There is a difference between passing a `pd.DataFrame` and the `column`  arg VS a `pd.Series`, i.e. `skbio.stats.distance.permanova(dm, meta['tumor'])`
```
In [6]: skbio.stats.distance.permanova(dm, meta, column='tumor')
Out[6]: 
method name               PERMANOVA
test statistic name        pseudo-F
sample size                      42
number of groups                  2
test statistic             1.076027
p-value                       0.299
number of permutations          999
Name: PERMANOVA results, dtype: object

In [5]: skbio.stats.distance.permanova(dm, meta['tumor'])
Out[5]: 
method name               PERMANOVA
test statistic name        pseudo-F
sample size                      42
number of groups                  2
test statistic             4.116198
p-value                       0.008
number of permutations          999
Name: PERMANOVA results, dtype: object

```

**Comment by sjanssen2** [comment]
I think it is really the `pd.Series` type of the grouping, as it seems the order of elements is not synced with DistanceMatrix ids prior to np.unique() here https://github.com/biocore/scikit-bio/blob/3519a38195215c06873580749dc9c55ecdfc9aa8/skbio/stats/distance/_base.py#L1297-L1305
Should the permanova method raise a warning that it cannot handle pd.Series but pd.DataFrames or should the `_preprocess_input_sng` function also handle pd.Series inputs?

**Comment by wasade** [comment]
Thank you, @sjanssen2. I think it would be best for the behavior to be consistent whether it's a series, list, tuple or dataframe with a column specified. Would you be able to issue a PR that implements a regression test, and fixes this behavior? It is _likely_ that `permdisp` is affected too.

**Comment by wasade** [comment]
@sjanssen2 this can be closed now right?

**Comment by sjanssen2** [comment]
yes, this issue should be solved by #1879
